appinventor/components/src/com/google/appinventor/components/runtime/CheckBox.java

```diff
@@ -70,4 +70,47 @@ public void Checked(boolean value) {
     view.invalidate();
   }
 
+  // START LinkedData
```

Comment:
Rather than putting this long here, you could put it in `ToggleBase` instead so that both `CheckBox` and `Switch` take it without duplicating the code.

Reply:
I swear it didn't work the first time when I moved it to ToggleBase (surely the first time someone has made that claim. Lol). But, I moved it back to ToggleBase and it's working now. 

(Building AppInventor took much longer before getting it to work on my linux box, so my frustration may have lead to some things falling through the cracks)